tools/cli-bootstrap/cli.bootstrap.py

```diff
@@ -0,0 +1,562 @@
+
+    except:
+        RoverSettings._DevMode = True
+        UnexpectedRoverException(sys.exc_info())
+
+
+
+def BuildNativeComponents(  coreclr_git_directory,
+                            corefx_git_directory,
+                            core_setup_git_directory,
+                            libuv_git_directory):
+    try:
+        RoverPrint(RoverMods.Blue('is building the .NET GitHub repositories.'))
+
+        # Build CoreCLR
+        # skipping non-essential for bootstrapping.
+        if 'coreclr' in RoverSettings.BuildSet:
+            RoverShellCall('./build.sh x64 release skiptests skipnuget', cwd=coreclr_git_directory)
+
+        # Build CoreFX Native Pieces
+        if 'corefx' in RoverSettings.BuildSet:
+            # at different points in the history of CoreFX there have been differing build behaviors, these conditionals
+            # cover that. However, if we find these differences, we will need to adapt.s
+            if path.exists(path.join(corefx_git_directory, 'src', 'Native', 'build-native.sh')):
+                RoverShellCall('./build-native.sh x64 release Linux --numProc 1', cwd="%s/src/Native"%(corefx_git_directory))
+            else:
+                RoverShellCall('./build.sh native x64 release', cwd=corefx_git_directory) 
+
+        # Build corehost from core-setup
+        # TODO: declare proper runtime id
+        # TODO: hostver?
+        # TODO: fxrver?
+        # TODO: policyver?
+        if 'core-setup' in RoverSettings.BuildSet:
+            RoverShellCall('./build.sh --arch x64 --rid %s --hostver 0.0.0 --fxrver 0.0.0 --policyver 0.0.0 --commithash %s'%(RoverSettings._Rid, RoverSettings.DotNetCommitHash), cwd="%s/src/corehost"%(core_setup_git_directory))
+
+        # Build libUV
+        if 'libuv' in RoverSettings.BuildSet:
+            RoverShellCall('./autogen.sh',     cwd=libuv_git_directory)
+            RoverShellCall('./configure',      cwd=libuv_git_directory)
+            RoverShellCall('make',             cwd=libuv_git_directory)
+
+    except:
+        RoverSettings._DevMode = True
+        UnexpectedRoverException(sys.exc_info())
+
+
+def PatchTarget(patchTarget_folder,
+                coreclr_bin_directory,
+                corefx_native_bin_directory,
+                core_setup_cli_bin_directory,
+                libuv_bin_directory):
+    try:
+        if RoverSettings.Patch:
+            RoverPrint(RoverMods.Blue('is patching %s'%(RoverMods.Yellow(patchTarget_folder))))
+                                
+            # replace native dotnet in the base directory
+            # from core_setup
+            RoverShellCall('cp dotnet %s'%(path.join(patchTarget_folder)), cwd='%s/exe/'%(core_setup_cli_bin_directory))
+
+            # replace native files in 'shared' folder.
+            # from coreclr
+            RoverShellCall('cp *so %s'%(RoverSettings.PatchTarget_Shared),                               cwd=coreclr_bin_directory) 
```

Comment:
We don't need all of the *so files from coreclr's output, only the ones that should be in the shared framework directory.